### PR TITLE
Fix/clipping2d issues

### PR DESCRIPTION
**Pull request by nd-novorender**
https://trello.com/c/JHrGNyVS/895-cross-section-related-fixes-changes

Many small changes/fixes based on feedback. See checklist in attached task.

### CODE REVIEW

File: src/features/clippingPlanes/clippingPlaneInteractions.tsx [hunk 1]
```diff
@@ -351,12 +357,21 @@ function PlaneMenu({
...
+                        <ListItemText>{t("crossSection")}</ListItemText>
+                    </MenuItem>
+                ) : (
+                    <MenuItem onClick={swapCamera}>
+                        <ListItemIcon>
+                            <featuresConfig.orthoCam.Icon fontSize="small" />
+                        </ListItemIcon>
+                        <ListItemText>{t("crossSection")}</ListItemText>
+                    </MenuItem>
+                )}
```
Comment (barabadzhi) [inline src/features/clippingPlanes/clippingPlaneInteractions.tsx:374]: ALT:
```
const handleClick = newDesign ? openCrossSection : swapCamera;
const Icon = newDesign ? featuresConfig.crossSection.Icon : featuresConfig.orthoCam.Icon;

return (
    <MenuItem onClick={handleClick}>
        <ListItemIcon>
            <Icon fontSize="small" />
        </ListItemIcon>
        <ListItemText>{t("crossSection")}</ListItemText>
    </MenuItem>
);
```

File: src/features/crossSection/clipping2d.tsx [hunk 2]
```diff
@@ -43,50 +42,52 @@ export function Clipping2d({ width, height }: { width: number; height: number })
...
-    const planeRef = useRef(plane);
-    planeRef.current = plane;
-
     useEffect(() => {
         drawObjectsRef.current = null;
         needRedrawRef.current = true;
-    }, [hidden, defaultVisibility, displaySettings.showLabels]);
+    }, [plane, hidden, defaultVisibility, displaySettings.showLabels]);
 
+    const planeNormalStr = plane ? plane.normalOffset.slice(0, 3).join(",") : null;
```
Comment (barabadzhi) [inline src/features/crossSection/clipping2d.tsx:87]: Memorize the value?

Comment (nd-novorender) [reply src/features/crossSection/clipping2d.tsx:87]: Memoization can be more expensive in this particular case

**Review by barabadzhi — APPROVED** [review]
LGTM

**Pull request merged**